sorting/random_pivot_quick_sort.cpp

```diff
@@ -0,0 +1,305 @@
+#include <cassert>      /// for assert
+#include <algorithm>   /// for std::is_sorted()
+#include <array>      /// for std::array
+#include <tuple>     /// for returning multiple values form a function at once
+
+
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+/**
+ * @brief Functions for the [Random Pivot Quick Sort](https://www.sanfoundry.com/cpp-program-implement-quick-sort-using-randomisation) implementation
+ * @namespace random_pivot_quick_sort
+ */
+    namespace random_pivot_quick_sort {
+        /**
+         * @brief Utility function to print the array
+         * @tparam T size of the array
+         * @param arr array used to print its content
+         * @returns void
+         * */
+        template<size_t T>
+        void showArray(std::array<int, T> arr) {
+            for (int i = 0; i < arr.size(); i++) {
+                std::cout << arr[i] << " ";
+            }
+            std::cout << std::endl;
+        }
+
+        /**
+         * @brief Takes the start and end indices of an array and returns a random integer between the range of those two
+         * for selecting pivot element.
+         *
+         * @param start The starting index.
+         * @param end The ending index.
+         * @returns int A random number between start and end index.
+         * */
+        int getRandomIndex(int start, int end) {
+            srand(time(nullptr)); // Initialize random number generator.
+            int randomPivotIndex = start + rand() % (end - start + 1);
+            return randomPivotIndex;
+        }
+
+        /**
+         * Nothing fancy, just a simple function to swap two elements in array.
+         * @tparam size size of the array to be passed as argument.
+         * @param arr The actual array.
+         * @param i An index of array.
+         * @param j Another index of the passed array
+         * @returns std::array<int, size> Same array with swapped elems.
+        */
+        template<size_t size>
+        std::array<int, size> swap(std::array<int, size> arr, int i, int j) {
+            // Nothing fancy, just a simple snippet to swap two numbers in an array.
+            int temp = arr[i];
+            arr[i] = arr[j];
+            arr[j] = temp;
+            return arr;
+        }
+
+        /**
+         * A partition function which handles the partition logic of quick sort.
+         * @tparam size size of the array to be passed as argument.
+         * @param start The start index of the passed array
+         * @param end The ending index of the passed array
+         * @returns std::tuple<int, std::array<int, size>> A tuple of pivot index and pivot sorted array.
+        */
+        template<size_t size>
+        std::tuple<int, std::array<int, size>> partition(std::array<int, size> arr, int start, int end) {
+
+            int pivot = arr[end];  // Randomly selected element will be here from caller function (quickSortRP()).
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code, too ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂